spec/components/dashboard_component_spec.rb

```diff
@@ -241,4 +241,64 @@
       end
     end
   end
+
+  describe "#view_applicants" do
+    let(:organisation) { create(:school) }
+    let(:vacancies) { [vacancy] }
+    let(:vacancy) do
+      create(:vacancy, :published, organisations: [organisation])
+    end
+
+    before do
+      create(:job_application, :status_submitted, vacancy: vacancy)
+      allow(vacancy).to receive(:allow_job_applications?).and_return(uses_either_native_or_uploaded_job_application_form)
+    end
+
+    context "when vacancy uses either native or uploaded job application form" do
+      let(:uses_either_native_or_uploaded_job_application_form) { true }
+
+      context "when the selected type is live" do
+        let(:selected_type) { :live }
+
+        it "renders the link to view applicants" do
+          render_inline(subject)
+          expect(page).to have_link(
+            I18n.t("jobs.manage.view_applicants", count: 1),
+            href: Rails.application.routes.url_helpers.organisation_job_job_applications_path(vacancy.id),
+          )
+        end
+      end
+
+      context "when the selected type is draft" do
+        let(:selected_type) { :draft }
+
+        context "and receive_applications != 'uploaded_form" do
```

Comment:
:question: do we need this context at all? It doesn't seem to modify anything on the `vacancy`  and from what I see on the component code, the `draft` state never adds the link to view applicants. 

So this context seems not to be correct?

Reply:
oops yes good spot 👍